Cython/Compiler/PyrexTypes.py

```diff
@@ -334,6 +337,29 @@ def _assign_from_py_code(self, source_code, result_code, error_pos, code,
             convert_call,
             code.error_goto_if(error_condition or self.error_condition(result_code), error_pos))
 
+    def _generate_xxxref_placeholder(self, *ignored_args, **ignored_kwds):
+        if self.needs_xxxref:
+            raise NotImplementedError("Ref-counting operation not yet implemented for type %s" %
+                                      self)
+        else:
+            return None
+
+    def _generate_xxxref_set_placeholder(self, cname, rhs_cname, *ignored_args, **ignored_kwds):
+        if self.needs_xxxref:
+            raise NotImplementedError("Ref-counting operation not yet implemented for type %s" %
+                                      self)
+        else:
+            return "%s = %s" % (cname, rhs_cname)
+
+    generate_incref = generate_xincref = generate_decref = generate_xdecref \
+        = generate_decref_clear = generate_xdecref_clear \
+        = generate_gotref = generate_xgotref = generate_giveref = generate_xgiveref \
+        = generate_nullcheck \
```

Comment:
This feels unrelated and doesn't seem to be doing the right thing. I would expect it to be used in a condition, and then, what is the correct fallback there? A condition needs _something_ to check.
I would just remove this method.

Reply:
This was added because it was useful for the fastcall args change - the types could fail to generate correctly and there was a way of testing it. The fallback should be `1` (i.e. basic types can never fail to generate) but right now it isn't.

I think I'd like to keep something like this, but I'll see if I can make it clearer.